src/otx/algo/visual_prompting/encoders/sam_image_encoder.py

```diff
@@ -14,50 +13,24 @@ class SAMImageEncoder(nn.Module):
 
     backbone_configs: ClassVar[dict] = {
```

Comment:
As I can see this SAMImageEncoder is a factory to return configured VIT backbones. Could we move it to VIT side?

![image](https://github.com/user-attachments/assets/e0315d85-8191-46cf-8863-30810453c5f9)

Reply:
Yes, but I have 2 concerns:
1. It looks better to distinguish the image encoder from backbones on SAM's module structure.
2. We can consider the case when moving backbones to common backbone if possible.

2 can be regarded as a long plan and 1 is a design choice, so I'd like to suggest keeping this version.